### DISCUSSION TITLE
네이버지도 프레임워크를 사용하고자 설치를 했는데, 빌드가 안 되는 이슈가 생겼습니다

### DISCUSSION THREAD

**Discussion by ubeeni** [DeveloperAcademy-POSTECH/.github#28] [Q&A]
### ⚠️ 문제
<img width="915" alt="스크린샷 2024-06-17 오전 11 20 25" src="https://github.com/DeveloperAcademy-POSTECH/.github/assets/69234788/1caa6b51-5fd2-4bb2-97c7-454d3c227de2">

### ❗️시도
- `Target > BuildSettings -> Build Options > User Script Sandboxing 옵션`을 `No`로 바꿔줌 
- 해결?!!!

<img width="915" alt="스크린샷 2024-06-17 오전 11 20 56" src="https://github.com/DeveloperAcademy-POSTECH/.github/assets/69234788/41eb5563-5e1b-4f9a-bbfa-66497b5ff331">

### ❓궁금한 점
- **SandBox란?** 간단히 이야기하면 앱의 접근을 제한시켜 시스템 리소스와 사용자 데이터를 보호하는 역할을 한다고 합니다. macOS 앱을 배포하려면 이 기능이 활성화되어 있어야 AppStore 등록이 가능하다고 하네요. 
- 그런데 이걸 No로 바꾸면 이슈는 해결이 되는데 이게 맞을까요?! 왜 될까요? macOS 앱을 쓸 땐 이 이슈가 발생하지 않을까요?

**Comment by moonkey48** [comment]
저도 처음보는 개념이어서 덕분에 조금 알아보는 시간을 가졌습니다 :)

먼저 Sandbox라는 개념 자체는 말씀하신 것처럼 앱의 접근 권한을 막아서 해킹이나 보안에 위험한 부분을 제어하기 위한 macOS의 접근제어 기술인 것 같습니다. 네이버 지도 API가 추가적인 script를 실행시키려고 Xcode 14에 나온 User Script Sandboxing을 풀어줘야 가능했던 기능이었나봅니다 ㅎㅎ 제가 설명하는 것보다 좋은 자료가 있어서 공유 드려요! 

https://hyunndyblog.tistory.com/149
https://indiestack.com/2023/06/xcode-build-script-sandboxing/
